tests/b2share_unit_tests/records/test_records_rest.py

```diff
@@ -388,3 +388,75 @@ def test_modify(status, user=None):
         test_modify(403, com_member)
         test_modify(200, com_admin)
         test_modify(200, admin)
+
+
+def test_delete_record(app, test_users, test_communities, login_user):
+    """Test record deletion through the REST API."""
+    uploaded_files = {
+        'myfile1.dat': b'contents1',
+        'myfile2.dat': b'contents2'
+    }
+    admin = test_users['admin']
+
+    with app.app_context():
+        creator = create_user('creator')
+        non_creator = create_user('non_creator')
+
+        record_data = generate_record_data()
+        with app.test_client() as client:
+            _, record_pid, record = create_record(
+                record_data, creator, files=uploaded_files
+            )
+
+        def test_delete(status, user=None):
+            with app.test_client() as client:
+                login_user(user, client)
+                headers = [('Accept', 'application/json')]
+                request_res = client.delete(
+                    url_for('b2share_records_rest.b2rec_item',
+                            pid_value=record_pid.pid_value),
+                    headers=headers)
+
+                assert request_res.status_code == status
+
+        test_delete(403, creator)
+        test_delete(403, non_creator)
+        test_delete(200, admin)
+
+        with app.test_client() as client:
+            login_user(creator, client)
```

Comment:
This part could have been factorized as it repeats 3 times nearly the same code.